core/src/main/scala/cats/data/Chain.scala

```diff
@@ -566,32 +566,26 @@ sealed abstract class Chain[+A] {
    * Returns the number of elements in this structure
    */
   final def length: Long = {
```

Comment:
I don't know if we have a test for this, but if not we should add that `chain.length == chain.toList.length.toLong`

Reply:
I'm really wondering how come `Chain#length` became `Long` rather than just `Int` (which is used for all other collections' length). Is there any chance that `Chain#length` can outgrow `Int` type?

Reply:
Yeah, it's annoying - it means `Chain` can never extend `Iterable` or `Seq`.

Reply:
sure it can, just cast to Int, but I don't think that's a good idea anyway. Make a wrapper and call `toSeq` and just implement the methods that way.

Why is the length an Int in scala? I think because java did this in java.util. Why did java.util do it? In those days computers were all 32 bits, so Long was more expensive (that isn't the case now, I don't think), also the idea of an arraylist with more than 2 billion items seemed crazy in the late 90s.

With scala List, there is nothing that I know of that prevents you from making a list more than 2 billion long: it is just a series of pointers. But if you do, I guess `length` will just throw or return junk.

In cats, we preferred Long for two reasons:

1. Long is as cheap as Int now.
2. We can actually imagine cases where Int is overflowed, but Long will basically never be overflowed because 2^63 is a gigantic number (we may live long enough to see computers with that much ram, but I doubt it).

So, if you want to implement `def toSeq: Seq[A]` on Chain, you can totally do it. The length being cast to `Int` is no more of a lie than when `List` does it, and we live with that all the time. I don't see the problem.

Reply:
> sure it can, just cast to Int

Not without return type polymorphism

Reply:
Good point, but the `toSeq` or `toIterable` without doing a deep copy does work.